File: codeForces/contests/959div2/E1.py
for _ in range(int(input())):
    k = int(input())
    sizes = []
    for _ in range(k):
        n = int(input())
        sizes.append(n)
        p = list(map(int, input().split()))
    sizes.sort()
    ans = 0
    while sizes:
        s = sizes.pop()
        if (1 << (s.bit_length()-1)) & ans != 0:
            ans |= (1 << s.bit_length()-1) - 1
            break
        ans |= s
    print(ans)

File: codeForces/contests/959div2/E1.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <cmath>

using namespace std;

int main() {
    int t;
    cin >> t;

    while (t--) {
        int k;
        cin >> k;
        vector<int> sizes;

        for (int i = 0; i < k; ++i) {
            int n;
            cin >> n;
            sizes.push_back(n);
            int val;
            for (int i = 1; i < n; ++ i) {
                cin >> val;
            }
        }

        sort(sizes.begin(), sizes.end());

        unsigned int ans = 0;
        while (!sizes.empty()) {
            int s = sizes.back();
            sizes.pop_back();
            
            unsigned int highest_bit = 1 << (int(log2(s)));

            if ((highest_bit & ans) != 0) {
                ans |= (highest_bit - 1);
                break;
            }
            ans |= s;
        }
        
        cout << ans << endl;
    }

    return 0;
}
